src/Swashbuckle.AspNetCore.Swagger/ISwaggerProvider.cs

```diff
@@ -13,6 +14,14 @@ public interface ISwaggerProvider
             string basePath = null);
     }
 
+    public interface IAsyncSwaggerProvider
+    {
+        Task<OpenApiDocument> GetSwaggerAsync(
+            string documentName,
+            string host = null,
+            string basePath = null);
+    }
+
```

Comment:
Independently of this change, I was wanting to change the `GetSwagger` signature to the following as it aligns more closely with Swagger/OpenAPI terminology.

```csharp
OpenApiDocument GetSwagger(string documentName, string serverUrl = null);
```

Now, seeing as we're introducing a _new_ interface anyway, I think it might be a good opportunity to introduce that preferred signature. Does that make sense?

Reply:
Furthermore, and on a similar note of opportunism, I wonder if we should call it `IOpenApiProvider` instead of `IAsyncSwaggerProvider`?

Reply:
Well, of course we could change the interface, but that sounds breaking. Is it a common use case to use ISwaggerProvider directly? I never did that, therefore doesn't hurt me :)
Would you change the CLI as well to accept --serverurl rather then --host and --basepath?
Of course just renaming IAsyncSwaggerProvider to IOpenApiDocumentProvider would unproblematic